ipadnsrecord: Fix documentation sections and agument spec

t-woerner opened the pull request:
ansible-test with ansible-2.14 is adding a lot of new tests to ensure that the documentation section and the agument spec is complete. Needed changes:

DOCUMENTATION section

- `type: str` needs to be set for string parameters
- `type: list` needs to be set for list parameters
- `elements: str` needs to be given for list of string parameters
- `type: string` is not valid and needs to be replaced by `type: str`
- `suboptions` instead of `options` needs to be used for dict parameters
- `authors` needs to be given with the github user also: `Name (@user)`
- `choices` needs to match `argument_spec`
- `description` needs to match parameter
- all parameters need to be defined

argument_spec

- `elements="str"` needs to be added to all list of string parameters
- `no_log=False` or `no_log=True` needs to be set for all parameters that have `key` in the name or for dicts also in one the key names
- `elements="dict"` needs to be added to all list of dict parameters

The `copyright` date is extended with `-2022`.

rjeffman commented on plugins/modules/ipadnsrecord.py:
```diff
@@ -273,7 +321,7 @@
         description: LOC Direction Latitude
         required: false
-        choices: ["N", "S"]
+        type: str
```
Shouldn't we keep the choices "N" and "S", as only them should be used (RFC 1876)?

t-woerner replied:
The `choices` are only in the docmentation section, but are missing in `argument_spec`. If you agree, then I can add them to `argument_spec`.

rjeffman commented on plugins/modules/ipadnsrecord.py:
```diff
@@ -289,7 +337,7 @@
         description: LOC Direction Longitude
         required: false
-        choices: ["E", "W"]
+        type: str
```
Shouldn't we keep the choices "E" and "W", as only them should be used (RFC 1876)

rjeffman replied:
oh... my bad. They should be on the argument_spec. Please add to both.

rjeffman commented on plugins/modules/ipadnsrecord.py:
```diff
@@ -641,7 +737,7 @@
     description: LOC Direction Latitude
     required: false
-    choices: ["N", "S"]
+    type: str
```
Shouldn't we keep the choices "N" and "S", as only them should be used (RFC 1876)?

rjeffman commented on plugins/modules/ipadnsrecord.py:
```diff
@@ -657,7 +753,7 @@
     description: LOC Direction Longitude
     required: false
-    choices: ["E", "W"]
+    type: str
```
Shouldn't we keep the choices "E" and "W", as only them should be used (RFC 1876)?

t-woerner commented:
I have added all the choices and also changed the element type to "dict" again to fix an issues in `tests/dnsrecord/env_cleanup.yml` in task `Ensure that "{{ safezone }}" dns records are absent`. Additional code has been added to remove all keys that have a None value from the dicts in `records` list. With this it behave like before with elements=None.

rjeffman reviewed (commented):
lgtm
